dotnet/src/webdriver/Platform.cs

```diff
@@ -84,138 +86,92 @@ public enum PlatformType
     /// </summary>
     public class Platform
     {
-        private static Platform current;
-        private PlatformType platformTypeValue;
-        private int major;
-        private int minor;
+        private static Platform? current;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="Platform"/> class for a specific platform type.
         /// </summary>
         /// <param name="typeValue">The platform type.</param>
         public Platform(PlatformType typeValue)
         {
-            this.platformTypeValue = typeValue;
+            this.PlatformType = typeValue;
         }
 
         private Platform()
         {
-            this.major = Environment.OSVersion.Version.Major;
-            this.minor = Environment.OSVersion.Version.Minor;
+            this.MajorVersion = Environment.OSVersion.Version.Major;
+            this.MinorVersion = Environment.OSVersion.Version.Minor;
 
             switch (Environment.OSVersion.Platform)
             {
                 case PlatformID.Win32NT:
-                    if (this.major == 5)
+                    if (this.MajorVersion == 5)
                     {
-                        this.platformTypeValue = PlatformType.XP;
+                        this.PlatformType = PlatformType.XP;
                     }
-                    else if (this.major == 6)
+                    else if (this.MajorVersion == 6)
                     {
-                        this.platformTypeValue = PlatformType.Vista;
+                        this.PlatformType = PlatformType.Vista;
                     }
                     else
                     {
-                        this.platformTypeValue = PlatformType.Windows;
+                        this.PlatformType = PlatformType.Windows;
                     }
 
                     break;
 
                 // Thanks to a bug in Mono Mac and Linux will be treated the same  https://bugzilla.novell.com/show_bug.cgi?id=515570 but adding this in case
                 case PlatformID.MacOSX:
-                    this.platformTypeValue = PlatformType.Mac;
+                    this.PlatformType = PlatformType.Mac;
                     break;
 
                 case PlatformID.Unix:
-                    this.platformTypeValue = PlatformType.Unix;
+                    this.PlatformType = PlatformType.Unix;
                     break;
             }
         }
 
         /// <summary>
         /// Gets the current platform.
         /// </summary>
-        public static Platform CurrentPlatform
-        {
-            get
-            {
-                if (current == null)
-                {
-                    current = new Platform();
-                }
-
-                return current;
-            }
-        }
+        public static Platform CurrentPlatform => current ??= new Platform();
```

Comment:
The only change I made here was using the modern `??=` syntax. If we want to replace this with `Lazy<Platform>` I can make that change.

Reply:
Minor thing, I don't know a strong rule. `Lazy<>` is overhead in terms of simplicity.